File: mobile/lib/services/reserva_service.dart
import '../api/api_client.dart';
import '../models/reserva.dart';

class ReservaService {
  static Future<Reserva> crear(ReservaCreate data) async {
    final resp = await ApiClient.post('/reservas/', body: data.toJson());
    return Reserva.fromJson(resp as Map<String, dynamic>);
  }

  static Future<List<Reserva>> misReservas() async {
    final resp = await ApiClient.get('/reservas/');
    return (resp as List)
        .map((e) => Reserva.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<Reserva> cancelar(int idReserva) async {
    final resp =
        await ApiClient.patch('/reservas/$idReserva/cancelar');
    return Reserva.fromJson(resp as Map<String, dynamic>);
  }
}

File: frontend/src/app/services/reserva.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';

import { environment } from '../../environments/environment';
import { Reserva, ReservaCreate } from '../models/reservas';

@Injectable({ providedIn: 'root' })
export class ReservaService {
  private http = inject(HttpClient);

  crear(data: ReservaCreate) {
    return this.http.post<Reserva>(`${environment.apiUrl}/reservas/`, data);
  }

  misReservas() {
    return this.http.get<Reserva[]>(`${environment.apiUrl}/reservas/`);
  }

  getPendientes() {
    return this.http.get<Reserva[]>(`${environment.apiUrl}/reservas/pendientes`);
  }

  cancelar(id: number) {
    return this.http.patch<Reserva>(`${environment.apiUrl}/reservas/${id}/cancelar`, {});
  }

  preparar(id: number) {
    return this.http.patch<Reserva>(`${environment.apiUrl}/reservas/${id}/preparar`, {});
  }

  completar(id: number) {
    return this.http.patch<Reserva>(`${environment.apiUrl}/reservas/${id}/completar`, {});
  }
}